Server/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```diff
@@ -70,374 +72,120 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Blip", "Element");
 }
 
-int CLuaBlipDefs::CreateBlip(lua_State* luaVM)
+std::variant<CBlip*, bool> CLuaBlipDefs::CreateBlip(lua_State* const luaVM, const CVector vecPosition, const std::optional<std::uint8_t> icon, std::optional<std::uint8_t> size, const std::optional<std::uint8_t> r, const std::optional<std::uint8_t> g, const std::optional<std::uint8_t> b, const std::optional<std::uint8_t> a, const std::optional<std::int16_t> ordering, const std::optional<std::uint16_t> visibleDistance, const std::optional<CElement*> visibleTo)
```

Comment:
tbf you shouldn't pass boolean to an element either way.
I think if you provide `nil` then it will leave it as an empty optional
Maybe optional variant?

Reply:
I think this is expected behavior if a different type is specified than required, in this case boolean for the element. In the case of nil, the error does not occur because it means there is no argument value